closed/adds/jdk/src/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -1101,145 +940,844 @@ private void listUsedProfile() {
+            for (String descKey : descKeys) {
+                System.out.println(descKey);
+            }
+            System.out.println();
+
+            // List providers.
+            System.out.println(profileToPrint + " Profile Providers:");
+            System.out.println("===============================================");
+            for (String provider : providers) {
+                System.out.println(provider);
             }
-            return deque.isEmpty();
+            System.out.println();
+
+            // List profile restrictions.
+            System.out.println(profileToPrint + " Profile Restrictions:");
+            System.out.println("==================================================");
+            for (String restriction : restrictions) {
+                System.out.println(restriction);
+            }
+            System.out.println();
         }
 
         /**
-         * Check if the input string is asterisk (*).
+         * Only set a property if the value is not null.
          *
-         * @param string input string for checking
-         * @return true if the input string is asterisk
+         * @param property      the property to be set
+         * @param propertyKey   the property key in the java.security file
+         * @return              whether the property was set
          */
-        private static boolean isAsterisk(String string) {
-            return "*".equals(string);
+        private boolean setProperty(String property, String propertyKey, List<String> allInfo) {
+            if (debug != null) {
+                debug.println("Setting property: " + property);
+            }
+            String value = securityProps.getProperty(propertyKey);
+            value = parseProperty(value);
+            String newValue = null;
+            if (value != null) {
+                // Add to info to create hash.
+                allInfo.add(propertyKey + "=" + value);
+
+                // Check if property overrides, adds to or removes from previous value.
+                String existingValue = profileProperties.get(property);
+                if (value.startsWith("+")) {
+                    if (!isPropertyAppendable(property)) {
+                        printStackTraceAndExit("Property '" + property + "' is not appendable.");
+                    } else {
+                        // Append additional values to property.
+                        value = value.substring(1).trim();
+
+                        // Take existing value of property into account, if applicable.
+                        if (existingValue == null) {
+                            printStackTraceAndExit("Property '" + property + "' does not exist in parent profile. Cannot append.");
+                        } else if (isBlank(existingValue)) {
+                            newValue = value;
+                        } else {
+                            newValue = (isBlank(value)) ? existingValue : existingValue + ", " + value;
```

Comment:
Please remove the redundant parentheses:
```java
                            newValue = isBlank(value) ? existingValue : existingValue + ", " + value;
```

Reply:
Removed.